File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/DemoUtility/UIUtility/DemoGetSetViewController.m
//
//  DemoGetSetViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//
/**
 *  A UI base view controller that is used for setting and getting a parameter of a component. A view controller for a component can inherit
 *  this class for convenience.
 */
#import "DemoGetSetViewController.h"

@interface DemoGetSetViewController ()

@end

@implementation DemoGetSetViewController

-(instancetype)init {
    return [super initWithNibName:@"DemoGetSetViewController"  bundle:[NSBundle mainBundle]];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // The getValueTextField is only used for displaying, not interaction is required.
    [self.getValueTextField setUserInteractionEnabled:NO];
}

- (IBAction)onGetButtonClicked:(id)sender {
}

- (IBAction)onSetButtonClicked:(id)sender {
}
@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/DemoUtility/UIUtility/DemoGetSetViewController.swift
//
//  DemoGetSetViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 12/17/15.
//  Copyright © 2015 DJI. All rights reserved.
//
import UIKit
class DemoGetSetViewController: DJIBaseViewController {
    @IBOutlet weak var getValueTextField: UITextField!
    @IBOutlet weak var setValueTextField: UITextField!
    @IBOutlet weak var getValueButton: UIButton!
    @IBOutlet weak var setValueButton: UIButton!
    @IBOutlet weak var rangeLabel: UILabel!

    @IBAction func onGetButtonClicked(_ sender: AnyObject) {
    }

    @IBAction func onSetButtonClicked(_ sender: AnyObject) {
    }

    init() {
        super.init(nibName: "DemoGetSetViewController", bundle: Bundle.main)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        self.getValueTextField.isUserInteractionEnabled = false
    }
}
